src/commands/msetnx.rs

```diff
@@ -0,0 +1,204 @@
+use std::sync::{Arc, Mutex};
+
+use bytes::Bytes;
+
+use crate::commands::executable::Executable;
+use crate::commands::{CommandParser, CommandParserError};
+use crate::frame::Frame;
+use crate::store::Store;
+use crate::Error;
+
+/// Sets the given keys to their respective values.
+/// Will not perform any operation at all even if just a single key already exists.
+///
+/// Ref: <https://redis.io/docs/latest/commands/msetnx/>
+#[derive(Debug, PartialEq)]
+pub struct Msetnx {
+    pub pairs: Vec<(String, Bytes)>,
+}
+
+impl Executable for Msetnx {
+    fn exec(self, store: Arc<Mutex<Store>>) -> Result<Frame, Error> {
+        if self.pairs.is_empty() {
+            return Ok(Frame::Error(
+                "ERR wrong number of arguments for command".to_string(),
+            ));
+        }
+
+        // NOTE:
+        // We could add some "transaction" logic that could be reverted.
+        // This way we wouldn't have to check on all the keys before setting them.
+        // If we found one that exists, we rollback and return 0.
```

Comment:
Not sure about this. I think what we gain on adding without checking we might lose on the rolling back operation.
I think your logic is simpler I like that.

Reply:
We should definitely analyze more what would be more performant before choosing such approach.